src/crypto/index.js

```diff
@@ -1088,7 +1093,16 @@ Crypto.prototype._backupPendingKeys = async function(limit) {
         undefined, undefined, this.backupInfo.version,
         {rooms: data},
     );
-    await this._cryptoStore.unmarkSessionsNeedingBackup(sessions);
+
+    remaining = await this._cryptoStore.doTxn(
+        'readwrite',
+        IndexedDBCryptoStore.STORE_BACKUP,
+        async (txn) => {
+            await this._cryptoStore.unmarkSessionsNeedingBackup(sessions, txn);
+            return this._cryptoStore.countSessionsNeedingBackup(txn);
```

Comment:
I'm not sure this will work reliably since afaik indexeddb will close the txn as soon as the callback function is done finishes synchronously, so if the promises impl doesn't resume the async function straight away, the txn will have been closed by the time it gets to `countSessionsNeedingBackup`.

Reply:
Thanks for pointing this out. I have reworked things to just use separate transactions, which to be honest is also easier to read (and just fine for this situation).